app/src/main/java/com/example/atsuko/simplewarikan/CostCalculator.kt

```diff
@@ -4,6 +4,9 @@ class CostCalculator {
 
 
     fun calculateCost(cost: String, peopleCount: String): String {
-        return (cost.toInt() / peopleCount.toInt()).toString()
+        if (!"""\d+(\.\d+)?""".toRegex().matches(cost)) return ""
```

Comment:
[wow] This is some regex magic 😮

Reply:
![](https://i.imgur.com/YsbKHg1.gif?noredirect)

Reply:
💯

Reply:
I want a それな stamp for Github...

I agree with rallat - regex looks like black magic (because it is), and so that can make it difficult to maintain